### PR TITLE
Logging of Deployment and Pod statuses

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**

/kind flake

**What does does this PR do / why we need it**:

Log Deployment and Pod status.
This will make it easier to debug issues like 
https://github.com/openshift/odo/issues/3256
https://github.com/openshift/odo/issues/2877

Because we will see a  reason why Deployment or Pod is failing

**Which issue(s) this PR fixes**:

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by kadel** [comment]
/retest
```
2020/08/27 14:09:41 error: artifacts container for pod integration-e2e-ipi-conf-aws unready: timed out waiting for the condition
2020/08/27 14:10:12 error: unable to signal to artifacts container to terminate in pod integration-e2e-ipi-conf-aws, triggering deletion: could not run remote command: unable to upgrade connection: container not found ("artifacts")
```

File: vendor/k8s.io/api/apps/v1/types.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 package v1
 
 import (
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (mik-dass) [inline vendor/k8s.io/api/apps/v1/types.go:20]: is this change required?

Comment (kadel) [reply vendor/k8s.io/api/apps/v1/types.go:20]: no, good catch. This is mistake

**Comment by kadel** [comment]
```
++ oc apply -n openshift -f https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json
error: unable to read URL "https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json", server reported 503 Backend is unhealthy, status code=503
```
/retest

**Review by mik-dass — APPROVED** [review]
/lgtm
